src/configuration.ts

```diff
@@ -120,8 +120,83 @@ export function loadConfigurationFromPath(configFilePath: string) {
         let fileData = fs.readFileSync(configFilePath, "utf8");
         // remove BOM if present
         fileData = fileData.replace(/^\uFEFF/, "");
-        return JSON.parse(fileData);
+        let config = JSON.parse(fileData) as IConfigFile;
+
+        let parentConfigNames = config.extends;
+        if (parentConfigNames) {
+            const configFileDir = path.dirname(configFilePath);
+            if (Array.isArray(parentConfigNames)) {
+                (<string[]>parentConfigNames).forEach(name => {
+                    const parentConfigFilePath = resolveConfigFilePath(name, configFileDir);
+
+                    let parentConfig = loadConfigurationFromPath(parentConfigFilePath);
+                    config = applyExtends(config, parentConfig);
+                });
+            } else {
+                const parentConfigFilePath = resolveConfigFilePath(<string>parentConfigNames, configFileDir);
+
+                let parentConfig = loadConfigurationFromPath(parentConfigFilePath);
+                config = applyExtends(config, parentConfig);
+            }
+
+            delete config.extends;
+        }
+
+        return config;
+    }
+}
+
+export interface IConfigFile {
+    extends: string | string[];
+    rulesDirectory: string | string[];
+    rules: any;
+}
+
+/**
+ * Resolve config file path.
+ * @var {string} relativeFilePath Relative path or
+ *   package name (tslint-config-X) or package short name (X)
+ */
+function resolveConfigFilePath(relativeFilePath: string, relativeTo: string) {
+    if (relativeFilePath.indexOf("/") === -1) {
+        relativeFilePath = `tslint-config-${relativeFilePath}`;
     }
+
+    if (relativeFilePath.indexOf("tslint-config-") === 0) {
```

Comment:
What do you mean by `require` here? Put `resolveConfigFilePath` to another file?